### DISCUSSION TITLE
Wallet is disconnected for a split second on page reload

### DISCUSSION THREAD

**Discussion by wahabshah21** [rainbow-me/rainbowkit#2125] [Q&A] [answered]
Hi I have this issue on one of my profile page. Where there is a check that if wallet isn't connected ui show a toast "wallet isnt connected" but after sometime say maybe after 1-2 seconds the wallet gets connected. Is there a way to handle this. Because previously only metamask was used but since integrating rainbow wallet i am getting this behavior.

**Comment by 0xmkh** [comment, marked as answer]
@wahabshah21 Can you check if it's also connecting / reconnecting rather than only looking for if it's disconnected only ? Wagmi has `useAccount` hook which could be helpful. Here is the documentation https://wagmi.sh/react/api/hooks/useAccount#status

**Reply by wahabshah21** [reply]
I will have a look Magiziz. So you suggestion is that I should handle what ever I am on reconnecting rather than just disconnected, right?

**Reply by 0xmkh** [reply]
Yes exactly

**Reply by wahabshah21** [reply]
It works, thanks
